lib/methods/dynamics/webApi/read.js

```diff
@@ -0,0 +1,279 @@
+   *
+   * @param {string} serviceId
+   * @return {Promise<Object>}
+   */
+  const readServiceRoles = async (serviceId) => {
+    const request = await readServiceRoles.buildRequest(serviceId)
+
+    return requestPromise(request).then(decodeResponse)
+  }
+
+  readServiceRoles.buildRequest = async (serviceId) => {
+    const params = {}
+
+    params['$filter'] = `_defra_lobservice_value eq ${serviceId}`
+
+    const headers = await buildHeaders()
+    const url = buildUrl(`/defra_lobserivceroles`, params)
+
+    return {
+      method: 'GET',
+      url,
+      headers
+    }
+  }
+
+  /**
+   * Fetches all enrolment records for the passed contact. Takes optional parameters for defining the roles and accountIds to filter by
+   *
+   * @param {string} contactId
+   * @param {Array|null} serviceRoleIds
+   * @param {Array|null} accountIds
+   * @return {Promise<Object>}
+   */
+  const readEnrolment = async (contactId, serviceRoleIds = null, accountIds = null) => {
+    const request = await readEnrolment.buildRequest(contactId, serviceRoleIds, accountIds)
+
+    return requestPromise(request).then(decodeResponse)
+  }
+
+  readEnrolment.buildRequest = async (contactId, serviceRoleIds = null, accountIds = null) => {
+    const params = {}
+
+    params['$filter'] = `_defra_serviceuser_value eq ${contactId}`
+
+    if (serviceRoleIds) {
+      params['$filter'] += ` and (`
+
+      params['$filter'] += serviceRoleIds.map(id => `_defra_servicerole_value eq ${id}`).join(' or ')
+
+      params['$filter'] += ')'
+    }
+
+    if (accountIds) {
+      params['$filter'] += ` and (`
+
+      params['$filter'] += accountIds.map(id => `_defra_organisation_value eq ${id}`).join(' or ')
+
+      params['$filter'] += ')'
+    }
+
+    const headers = await buildHeaders()
+    const url = buildUrl(`/defra_lobserviceuserlinks`, params)
+
+    return {
+      method: 'GET',
+      url,
+      headers
+    }
+  }
+
+  /**
+   *
+   * @param {string} serviceId
+   * @param {string} b2cObjectId
+   * @return {Promise<{roles: Array, mappings: Array}>}
+   */
+  const readServiceEnrolment = async (serviceId, b2cObjectId) => {
+    const [
+      { value: serviceRoles } = {},
+      contact
+    ] = await Promise.all([
+      readServiceRoles(serviceId),
+      readContacts({ b2cObjectId })
+    ])
+
+    const { dynamicsContactId: contactId } = contact[0]
+
+    if (!contact || !contact.length) {
+      throw Error(`Contact record not found for b2cobjectid ${b2cObjectId}`)
+    }
+
+    const enrolment = await readEnrolment(contactId, serviceRoles.map(role => role.defra_lobserivceroleid))
+
+    return readServiceEnrolment.parseResponse(enrolment)
+  }
+
+  readServiceEnrolment.parseResponse = (enrolment) => {
+    const response = {
+      roles: [],
+      mappings: []
+    }
+
+    enrolment.value.forEach(enrol => {
+      response.roles.push(`${enrol._defra_organisation_value}:${enrol._defra_servicerole_value}:${enrol.defra_enrolmentstatus}`)
+
```

Comment:
Blank lines?

Reply:
Just indicating separating between pushing to roles and mappings